group-generators/helpers/data-providers/eas/index.ts

```diff
@@ -0,0 +1,123 @@
+import { GraphQLClient } from "graphql-request";
+import { GetAttestationParams, QueryParams } from "./types";
+import { FetchedData } from "topics/group";
+
+export const defaultLimit = 100;
+
+export class EthereumAttestationServiceProvider {
```

Comment:
the base Uri is actually configurable from the params as you can query EAS on the different networks (sepolia, arbitrum, mainnet (default)). So it would require instantiation to get updated. Which is all generic for all providers at the moment?

Reply:
I think you can use the setEndpoint() method (`this.graphQLClient.setEndpoint("");`) to change the URI of the client can't you?
It's used in some of our Providers, so I wanted to standardize this in all our Providers that use graphql requests.
But if it's not possible then it's not so bad ;)

Reply:
Nice. Didn't see that. I'm all up for standardizing the data providers.

In fact, I spoke briefly about this with @leosayous21 around ETHport.
With a proper set of interface/base classes, I think you could dynamically fetch and compose most of these data providers, without needing for the ./index and interface-schemas.

Maybe not a priority, but would be nice for the future :)

Reply:
Yes you're right!
Agree with you that it's a bit of a struggle to create a data provider, the devx is quite bad, we definitely need to improve this in the futur yes!